### ISSUE TITLE
`RedisCluster` has a problem: expected engine to be one of ["redis"], got valkey

### ISSUE THREAD

**Issue by vimtor** [anomalyco/sst#5210]
I'm getting this error after upgrading from SST `3.3.59` to `3.4.3`:

```
✕  Failed    
   RedisCluster aws:elasticache:ReplicationGroup
   aws:elasticache/replicationGroup:ReplicationGroup resource 'RedisCluster' has a problem: expected engine to be one of ["redis"], got valkey. Examine values at 'RedisCluster.engine'.
```

The infra component did not change:

```ts
new sst.aws.Redis("Redis", {
  vpc,
  engine: "valkey",
  instance: isProduction ? "t4g.medium" : "t4g.micro",
})
```

**Comment by vimtor** [comment]
![image](https://github.com/user-attachments/assets/97aeb185-e861-4019-91f3-f61cff92a403)

> Me waiting to save 33% on my ElastiCache bill

**Comment by fwang** [comment]
Fixed since v3.4.7.

^ context: The issue was that SST used an older version of the AWS provider which didn't have the `valkey` option. The reason this error didn't show up in older versions (ie. v3.3.59) was b/c user's app always used the latest AWS provider. Around v3.4.3, we made the user's app use the same version that SST uses. Because SST was using an older provider version, it didn't have the `valkey` option.
